### DISCUSSION TITLE
Metrics with filtering

### DISCUSSION THREAD

**Discussion by phortx** [naymspace/backpex#560] [Q&A]
Hi,

is it possible to filter a query for a metric? For example: Display the amount of users with active subscriptions etc.

**Comment by Flo0807** [comment]
Hey,

we know that filters could be more flexible, but the `select: dynamic()` is very powerful. 

For example, if your User schema has an `active_subscription` boolean field that shows whether a user has a subscription or not, you can add the following metric:

```elixir
@impl Backpex.LiveResource
def metrics do
  [
    active_subscription_count: %{
      module: Backpex.Metrics.Value,
      label: "Active Subscription Users",
      class: "lg:w-1/4",
      select: dynamic([u], count(fragment("CASE WHEN ? = TRUE THEN 1 ELSE NULL END", u.active_subscription))),
      format: fn value ->
        Integer.to_string(value)
      end
    }
  ]
end
```

I added a metric that shows the number of published posts in our demo application to see it in action: https://github.com/naymspace/backpex/pull/561 (not released yet)

**Comment by phortx** [comment]
Ah that was the missing puzzle piece for me. Thanks for your answer again :)
